src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
Ah, I see now. Thanks for typing up the examples on coderpad. This is more of my C++ inexperience coming out :).